core/common/src/PlanarClipMask.ts

```diff
@@ -0,0 +1,149 @@
+ * clipped by by any model with a higher priority.  The enumerated values are the defaults and may be overriden by setting
+ * [[PlanarClipMask]].priority.
+ * @see [[PlanarClipMaskProps]]
+ * @see [[PlanarClipMaskSettings]]
+ *  @beta
+ */
+export enum PlanarClipMaskPriority {
+  BackgroundMap = -2048,
+  GlobalRealityModel = -1024,
+  RealityModel = 0,
+  BIM = 2048,
+  Maximum = 4096,
+}
+
+/** JSON representation of the [[PlanarClipMask]]  properties.
+ * @beta
+ */
+export interface  PlanarClipMaskProps {
+  /** The mode that controls how the mask geometry is collected */
+  mode: PlanarClipMaskMode;
+  /** The model IDs for the mask geometry.  If omitted then the models viewed in the masked model viewport are used */
+  modelIds?: CompressedId64Set;
+  /** The SubCategory ids if mode is [[PlanarClipMaskMode.IncludeSubCategorie]] or element Ids if mode is [[PlanarClipMaskMode.IncludeElements] or [[PlanarClipMaskMode.ExcludeElements]] */
+  subCategoryOrElementIds?: CompressedId64Set;
+  /** Priority value if mode is [[PlanarClipMaskMode.Priority]]. */
+  priority?: number;
+  /** A value between 0 and 1 indicating mask transparency - A transparency of 0 (the default) indicates complete masking.  1 is completely transparent (no masking). */
+  transparency?: number;
+}
+
+/** Normalized representation of [[PlanarClipMaskProps]] with validation and default values applied.
+ * @beta
+ */
+export class PlanarClipMaskSettings {
+  /** The mode that controls how the mask geometry is collected */
+  public readonly mode: PlanarClipMaskMode = PlanarClipMaskMode.None;
+  /** The model IDs for the mask geometry.  If omitted then the models viewed in the masked model viewport are used */
+  public readonly modelIds?: CompressedId64Set;
+  /** The SubCategory ids if mode is [[PlanarClipMaskMode.IncludeSubCategorie]] or element Ids if mode is [[PlanarClipMaskMode.IncludeElements] or [[PlanarClipMaskMode.ExcludeElements]] */
+  public readonly subCategoryOrElementIds?: CompressedId64Set;
+  /** Priority value if mode is [[PlanarClipMaskMode.Priority]]. */
+  public readonly priority?: number;
+  /** A value between 0 and 1 indicating am override for mask transparency - A transparency of 0 (the default) indicates complete masking.  1 is completely transparent (no masking).  If no transparency is defined then the transparency of the mask elements are used */
+  public readonly transparency?: number;
+
+  /** Create a new [[PlanarClipMaskSettings]] object from a JSON object */
+  public static fromJSON(json?: PlanarClipMaskProps): PlanarClipMaskSettings {
+    if (!json)
+      return this.defaults;
+
+    return new PlanarClipMaskSettings(json.mode, json.transparency, json.modelIds, json.subCategoryOrElementIds, json.priority);
+  }
+  /** Create a [[PlanarClipMaskAettings]] object */
+  public static create(mode: PlanarClipMaskMode, modelIds?: Id64Set, subCategoryOrElementIds?: Id64Set, transparency?: number): PlanarClipMaskSettings | undefined {
```

Comment:
Awkward to have `create` function that accepts a bunch of args that aren't relevant to each mode. Missing `priority` option for priority mode?

Reply:
A separate method (createByPriority) is included for creating by priority... Are you advocating additional separate creation methods to reduce the unused args?

Reply:
Not necessarily, it just stuck out to me. Seems like `fromJSON` would be easier to use - only the args you care about need to be specified.

Reply:
I'll look again -- originally had fromJSON but then caller had to deal with compressing the IDs.  (create handles that for them).

Reply:
I think what I have is fine...  the create method handles compressing the IDs